src/FSharp.Control.Websockets.TPL/FSharp.Control.Websockets.TPL.fs

```diff
@@ -147,19 +157,19 @@ module ThreadSafeWebsocket =
         }
 
     let sendMessage (wsts : ThreadSafeWebSocket) cancellationToken bufferSize messageType stream = task {
-        let reply = new TaskCompletionSource<unit>()
+        let reply = new TaskCompletionSource<_>()
         let msg = Send(cancellationToken,bufferSize, messageType, stream, reply)
         let! accepted = wsts.sendChannel.SendAsync msg
-        do! reply.Task
+        return! reply.Task
     }
 
     let sendMessageAsUTF8(wsts : ThreadSafeWebSocket) cancellationToken (text : string) = task {
         use stream = IO.MemoryStream.UTF8toMemoryStream text
-        do! sendMessage wsts cancellationToken Websocket.defaultBufferSize WebSocketMessageType.Text stream
+        return! sendMessage wsts cancellationToken Websocket.defaultBufferSize WebSocketMessageType.Text stream
     }
 
     let receiveMessage (wsts : ThreadSafeWebSocket) cancellationToken bufferSize messageType stream = task {
-        let reply = new TaskCompletionSource<unit>()
+        let reply = new TaskCompletionSource<_>()
```

Comment:
This this has changed now, the `reply.Task` needs to be a `return!`.  Then in the `receiveMessageUTF8`, it needs to do a pattern match against the Result and return `Result<string,exn>`

Reply:
fixed 👍